### DISCUSSION TITLE
Does Reader View circumvent Javascript blocking?

### DISCUSSION THREAD

**Discussion by kekkoudesu** [uBlockOrigin/uBlock-issues#2623] [Q&A] [answered]
This is in regard to Firefox.

I see that uBlock Origin does not operate on `reader:about-scheme` unless you add specific rules. I was wondering if Reader View ran whatever Javascript was on the page to get access to any images. On the normal webpage, you can't view the images because they require Javascript to run. I have Javascript disabled by default. If I go to Reader View, I can see the images...which leads me to assume that the Javascript on the page is being run to get access to those images, even though I have disabled Javascript by default in uBlock Origin.

Is this the case? For security reasons, I disable Javascript by default. I'm wondering if I need to add these rules so Reader View does not run with Javascript enabled.

**Comment by ghost** [comment]
Sample?

**Reply by kekkoudesu** [reply]
Here's one: https://www.uswitch.com/mobiles/studies/mobile-statistics/

**Reply by gorhill** [reply]
I don't see anything in the logger being reported when using reader view on that page, so it does seem reader view bypasses uBO.

---

The full URL when in reader mode is:

    about:reader?url=https%3A%2F%2Fwww.uswitch.com%2Fmobiles%2Fstudies%2Fmobile-statistics%2F

**Reply by kekkoudesu** [reply]
I see. So it seems it would be best to avoid Reader View if I was concerned about running the Javascript on a webpage.

**Reply by ghost** [reply]
To me it looks like internal script is used: https://searchfox.org/mozilla-central/rev/c7504aeccf60d3cea1e5b62a024a417d1482267b/toolkit/components/reader/Readability.js#1973

**Reply by gorhill** [reply, marked as answer]
According to the code comment, it appears scripting is disabled in reader view? Debugger is blank in dev tools. Also, the code you link to looks up `[data-src]`, so I guess that is why the images load properly.

Here: https://searchfox.org/mozilla-central/rev/c7504aeccf60d3cea1e5b62a024a417d1482267b/toolkit/components/reader/Readability.js#1973
